src/app/core/internal/view/CircuitView.ts

```diff
@@ -1,17 +1,174 @@
-import {CircuitInternal} from "../impl/CircuitInternal";
+import {SVGDrawing} from "svg2canvas";
 
+import {V, Vector} from "Vector";
 
-export class CircuitView {
-    private readonly circuit: CircuitInternal;
+import {BezierCurve} from "math/BezierCurve";
+import {Matrix2x3}   from "math/Matrix";
+import {Transform}   from "math/Transform";
 
-    private canvas: HTMLCanvasElement;
+import {GUID}                                from "..";
+import {CircuitInternal}                     from "../impl/CircuitInternal";
+import {SelectionsManager}                   from "../impl/SelectionsManager";
+import {Assembler}                           from "./Assembler";
+import {Prims}                               from "./Prim";
+import {RenderGrid}                          from "./rendering/renderers/GridRenderer";
+import {RenderHelper}                        from "./rendering/RenderHelper";
+import {DefaultRenderOptions, RenderOptions} from "./rendering/RenderOptions";
+import {RenderScheduler}                     from "./rendering/RenderScheduler";
+import {PortPos}                             from "./PortAssembler";
 
-    public constructor(circuit: CircuitInternal, canvas: HTMLCanvasElement) {
+
+export abstract class CircuitView {
+    public readonly circuit: CircuitInternal;
+    public readonly selections: SelectionsManager;
+
+    public readonly options: RenderOptions;
+
+    public readonly scheduler: RenderScheduler;
+    public readonly renderer: RenderHelper;
+
+    public cameraMat: Matrix2x3;
+
+    public componentTransforms: Map<GUID, Transform>;
+    public componentPrims: Map<GUID, Prims>;
+
+    public localPortPositions: Map<GUID, PortPos>;
+    public portPositions: Map<GUID, PortPos>;
+    public portPrims: Map<GUID, Prims>;
+
+    public wireCurves: Map<GUID, BezierCurve>;
+    public wirePrims: Map<GUID, Prims>;
+
+    public constructor(circuit: CircuitInternal, selections: SelectionsManager) {
         this.circuit = circuit;
-        this.canvas = canvas;
+        this.selections = selections;
+
+        this.options = new DefaultRenderOptions();
+
+        this.scheduler = new RenderScheduler();
+        this.renderer = new RenderHelper();
+
+        this.cameraMat = this.calcCameraMat();
+
+        this.componentTransforms = new Map();
+        this.componentPrims = new Map();
+
+        this.localPortPositions = new Map();
+        this.portPositions = new Map();
+        this.portPrims = new Map();
+
+        this.wireCurves = new Map();
+        this.wirePrims = new Map();
+
+        this.scheduler.subscribe(() => this.render());
+
+        this.circuit.subscribe((ev) => {
+            // TODO: use event
```

Comment:
TODOs like this should include a note about when this should be done. It is a requirement for this pr? A requirement for merging into master? Something that can wait for beyond that?

If you want to get real fancy with it, we can use this[ unicorn linting rule](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/expiring-todo-comments.md) but I don't think it is strictly necessary

(same goes for other TODOs on this file)

Reply:
Yeah... you're definitely out-ing me here, I'll try and come up with a standard

Reply:
Okay, standard is kinda-set:
`TODO[the-branch-to-deal-with-this-in](the-person-in-charge)`, i.e.
`TODO[model_refactor](leon)` means this TODO needs to be dealt with before `model_refactor` can be merged, and Leon is in charge of it